packages/components/data-table-manager/src/data-table-manager.story.js

```diff
@@ -293,6 +300,122 @@ storiesOf('Components|DataTable', module)
+                  appearance="quiet"
+                  value={phoneNumberTextColor}
+                  onChange={(event) =>
+                    setPhoneNumberTextColor(event.target.value)
+                  }
+                  options={[
+                    { value: 'black', label: 'black' },
+                    { value: 'red', label: 'Red' },
+                    { value: 'green', label: 'Green' },
+                  ]}
+                />
+              </Spacings.Stack>
+            </Grid.Item>
+            <Grid.Item>
+              <Spacings.Stack scale={'l'}>
+                <div>Toggle display</div>
+                <RadioInput.Group
+                  id="toggle-display"
+                  name="toggle-display"
+                  value={radioInputValue}
+                  onChange={(event) => setRadioInputValue(event.target.value)}
+                >
+                  <RadioInput.Option value={'show'}>
+                    Show texts for a given column or row
+                  </RadioInput.Option>
+                  <RadioInput.Option value={'hide'}>
+                    Hide texts for a given column or row
+                  </RadioInput.Option>
+                </RadioInput.Group>
+              </Spacings.Stack>
+            </Grid.Item>
+          </Grid>
+        </>
+      );
+    };
+
+    const SecondCustomComponent = () => {
+      return (
+        <>
+          <Grid
+            gridGap={designTokens.spacing30}
+            gridTemplateColumns="repeat(2, 1fr)"
+          >
+            <Grid.Item>
+              <Spacings.Stack scale={'l'}>
+                <div>Choose Image size</div>
+                <SelectInput
+                  value={imageSize}
+                  onChange={(event) => setImageSize(event.target.value)}
+                  options={[
+                    { value: 'large', label: 'Large' },
+                    { value: 'medium', label: 'Medium' },
+                    { value: 'small', label: 'Small' },
+                  ]}
+                />
+              </Spacings.Stack>
+            </Grid.Item>
+            <Grid.Item>
+              <Spacings.Stack scale={'l'}>
+                <div>Toggle display</div>
+                <RadioInput.Group
+                  id="toggle-display"
+                  name="toggle-display"
+                  value={radioInputValue}
+                  onChange={(event) => setRadioInputValue(event.target.value)}
+                >
+                  <RadioInput.Option value={'show'}>
+                    Show texts for a given column or row
+                  </RadioInput.Option>
+                  <RadioInput.Option value={'hide'}>
+                    Hide texts for a given column or row
+                  </RadioInput.Option>
+                </RadioInput.Group>
+              </Spacings.Stack>
+            </Grid.Item>
+          </Grid>
+        </>
+      );
+    };
+
+    const customSettings = [
+      {
+        id: 'customSettings1',
```

Comment:
The payload can take as much settings as the user wants to access in the nested rows.